api_v3/lib/types/serverNode/base/KalturaServerNodeArray.php

```diff
@@ -3,15 +3,15 @@
  * @package api
  * @subpackage objects
  */
-class KalturaEdgeServerArray extends KalturaTypedArray
+class KalturaServerNodeArray extends KalturaTypedArray
 {
 	public static function fromDbArray(array $arr, KalturaDetachedResponseProfile $responseProfile = null)
 	{
-		$newArr = new KalturaEdgeServerArray();
+		$newArr = new KalturaServerNodeArray();
 		foreach($arr as $obj)
 		{
 		    /* @var $obj StorageProfile */
-			$nObj = new KalturaEdgeServer();
+			$nObj = KalturaServerNode::getInstance($obj);
```

Comment:
KalturaServerNode::getInstance expecting also to get the $responseProfile, and fromObject is not required after it.